### ISSUE TITLE
[Feature Request] - Improved Testing Mode

### ISSUE THREAD

**Issue by ChristianSilvermoon** [Wohlstand/TheXTech#359]
I feel like this would be a **huge** improvement.

## General Improvements In Level
Show current Section Number on Pause Screen, or transparent in corner

## Episode Testing Mode
World Maps can be *super* difficult to make due to lack of easy testing. It's always been a major pain point for me.

The experience making maps in Episodes as resulted in dozens of invalid player positions, broken save files, etc.

It would be nice if an Episode could be started in a sort of Testing Mode.

Testing Mode would play like normal, but with its own dedicated 'Editor' Save for testing purposes.

When Pausing the Game on a level square you could get the following new Menu Options:
* Level Paths
* * In this menu you can enable/disable paths based on the levels unlocks (IE 'beat' or 'unbeat' the level a certain way)
* Level Stars
* * You can enable/or disable certain stars having been collected within the level
* Reset Editing Save (Restarts Episode with fresh save, overwriting the old editor save)
* Return to Start (Return to Start Tile)
* Autostart Level (Play the auto Start level)

Allow player to have 0 lives, but do not game-over / return to title

Possible, magic hand support for real-time world map editing? :thinking:

### Episode Testing World Map Menu Mock-up Screenshot
An example of what the menu could look like
![image](https://user-images.githubusercontent.com/30162808/180821855-d2043ebb-0c9b-4c49-aeed-a7bab6aac4c7.png)
Green Paths: Open
Red Paths: Closed
Gray Paths: Unavailable For This Level

**Comment by ChristianSilvermoon** [comment]
> > Gray Paths: Unavailable For This Level
> 
> Does unavailable mean "can't be opened from here" or "no path here"?

I want to say probably both?

**Comment by ds-sloth** [comment]
I agree that it will be useful to make a world test mode eventually, and I agree that it should allow you to effectively avoid playing any levels but test going through the full episode.

I propose that there be a special "Level Info Window" that is opened when you press the action key during test mode. This window would display which Exit NPCs, Powerups, and Level/Map Warps occur in the level. It would also allow the player to simulate taking any of the Level or Map Warps (where the Level Warps would lead to the target level's Level Info Window).

As always, there would be an option to Restart (world, now) in the pause menu.

How close would this feature (alone) come towards addressing your goals here?

**Comment by ChristianSilvermoon** [comment]
> I agree that it will be useful to make a world test mode eventually, and I agree that it should allow you to effectively avoid playing any levels but test going through the full episode.
> 
> I propose that there be a special "Level Info Window" that is opened when you press the action key during test mode. This window would display which Exit NPCs, Powerups, and Level/Map Warps occur in the level. It would also allow the player to simulate taking any of the Level or Map Warps (where the Level Warps would lead to the target level's Level Info Window).
> 
> As always, there would be an option to Restart (world, now) in the pause menu.
> 
> How close would this feature (alone) come towards addressing your goals here?

It's a better solution than my idea of stuffing everything in the pause menu ^^

It puts the situational settings into context with their situation :P
